### ISSUE TITLE
"Cannot calculate expiration date." on custom duration use

### ISSUE THREAD

**Issue by Berbe** [PrivateBin/PrivateBin#815]
## Steps to reproduce
Set `3days = 259200` in `[expire_options]` configuration section

### What happens
` Cannot calculate expiration date.` displayed when a paste has been generated

### What should happen
No error message

## Basic information
**PrivateBin version:** 1.3.5

**Comment by rugk** [comment]
We had a similar error in the past, but that should have been fixed already. See https://github.com/PrivateBin/PrivateBin/issues/597 :thinking: 

In any case, thanks for reporting it.

**Comment by elrido** [comment]
The keyword needs to be `3day`, not `3days`. The calculation mechanism supports the following keywords:
https://github.com/PrivateBin/PrivateBin/blob/5f2daa5cd6e609bd88ef95e0c2b3ab27a6c75298/js/privatebin.js#L330-L332

**Comment by rugk** [comment]
So as it seems this question has been answered and I thus close this issue. If anyone has further questions on that topic, feel free to comment here, again.

**Comment by Berbe** [comment]
@elrido
> The keyword needs to be `3day`, not `3days`.

This corrected the problem indeed. Thanks!

@rugk @elrido
However, looking at the code, the application is *not* supposed to fail, but rather fall back to computing the duration in seconds, which in my case, would be 3 seconds.

1. If it worked, how would the user be notified of the _actual_ paste lifespan?
2. It *doesn't* work, as returning [the first part of the match](https://github.com/PrivateBin/PrivateBin/blob/1.3.5/js/privatebin.js#L361) returns a `string`, not an [expected `number`](https://github.com/PrivateBin/PrivateBin/blob/1.3.5/js/privatebin.js#L552).
   1. There is no error handling preventing the paste to be generated. What is the duration of the paste? Infinite?
   2. The application runs fine, only when a paste is being generated the errors occurs, and only on generating the email link. Looks to me as if configuration shall be validated beforehand to avoid harm, like everlasting pastes left behind...

**Comment by rugk** [comment]
Oh yeah, seems you have some good points there… :thinking:

**Comment by elrido** [comment]
The created paste will send the string and the backend will use the number of seconds given in the configuration file and not actually calculate anything. In your case that would be 259200 seconds.

This is a configuration value - it is up to the admin to configure a valid value. From the backend perspective, you can also configure a random string with some integer, but the frontend won't be able to display it. Feel free to raise a PR with improvements for the frontend.

**Comment by Berbe** [comment]
> This is a configuration value - it is up to the admin to configure a valid value.

I beg to differ : it is up to the application to validate loaded configuration. Digesting an incorrect one and allowing itself to run in a partially broken (hence hard to diagnose) state is utterly dangerous.

> Feel free to raise a PR with improvements for the frontend.

Even though not being a developer, I usually try do that to the best of my abilities.
Given the fact the bug is triggered very late, in a mail sending function, I have no clear idea on what to do with it. I haven't looked at the code loading the settings to see what could be done validation-wise; might give that a try during some allocated free time down the line.

**Comment by rugk** [comment]
> it is up to the application to validate loaded configuration. Digesting an incorrect one and allowing itself to run in a partially broken (hence hard to diagnose) state is utterly dangerous.

I agree that some error or so could be logged on startup or even to refuse to start.
So IMHO a PR on improving that is a good idea.
